### PR TITLE
Mini Challenge #1 and #2 - Core concepts, styling, and testing.

### PR DESCRIPTION
**Pull request by Gonzz-96**
This PR contains the code for the Mini Challenges #1 and #2 of the Wizeline Internal React Bootcamp.

Questions to be answered:

### 1. Using create-react-app, what do we need to set up for testing?
Nothing. The `create-react-app` command already include the `@testing-library/react` in its dependencies.. This library will detect the `*.spec.js(x)`  or `*.test.js(x)` files, and run the test suits in them.

### 2. What components are worth to test in your development?
In the case of unit testing, I'd say that the most isolated components, since they are more easily tested since they don't have almost any dependency in other part of the code. This makes testing really easy.

### 3. Can you apply TDD once you already created components?
No. In a few words, the TDD cycle requires to write some testing code first (even when the code won't run), then write the code that makes the testing code run, and finally refactor your production code. If a component is already created, it won't fit in the TDD cycle.

### CODE REVIEW

File: src/components/VideoCard/styled.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const Title = styled.p`
+  margin: 10px;
+  font-size: 20px;
+`;
+
+const Description = styled.p`
+  margin: 0px 10px 10px 10px;
+  font-size: 14px;
+  color: #8e8e8e;
+`;
```
Comment (PacoMojica) [inline src/components/VideoCard/styled.js:29]: Although in the form you said you weren’t talented as designer, this is a great job, it's simple but looks great 👍

File: src/components/VideoCard/VideoCard.component.jsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { CardContainer, Thumbnail, Title, Description } from './styled';
+
+const VideoCard = ({ title, description, thumbnail, alt = 'Youtube Video' }) => {
+  return (
+    <CardContainer>
+      <Thumbnail src={thumbnail} alt={alt} />
+      <Title>{title}</Title>
+      <Description>{description}</Description>
+    </CardContainer>
```
Comment (PacoMojica) [inline src/components/VideoCard/VideoCard.component.jsx:10]: Good, small and concise

File: src/pages/Home/Home.page.jsx [hunk 3]
```diff
@@ -1,39 +1,34 @@
...
-          </span>
-        </>
-      ) : (
-        <Link to="/login">let me in →</Link>
-      )}
-    </section>
+    <>
+      <Header />
+      <Title>Welcome to my minichallenge :)</Title>
+      <GridContainer>{getVideoCardsFromList(items)}</GridContainer>
```
Comment (PacoMojica) [inline src/pages/Home/Home.page.jsx]: Here you can do something like:
```jsx
{items.map((item) => {
  const { title, description, thumbnails } = item.snippet;
  return (
    <VideoCard title={title} description={description} thumbnail={thumbnail} key={someUniquevalue} />
  );
})}
```

You can use the function, but it is not that common, also, when looping throughout an array of items, remember that react requires the `key` prop to work properly, if you omit the `key` you may get some bugs that are not easy to identify

File: src/pages/Home/styled.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    grid-template-columns: 25% 25% 25%;
+  }
+
+  @media screen and (max-width: 1100px) {
+    grid-template-columns: 30% 30%;
+  }
+
+  @media screen and (max-width: 900px) {
+    grid-template-columns: 80%;
+  }
```
Comment (PacoMojica) [inline src/pages/Home/styled.js:26]: Some bonus points!!! 😃

**Review by PacoMojica — COMMENTED** [review]
Great! Keep the good work coming 👍😁

File: src/components/VideoCard/styled.js (hunk 1 above)
Comment (Gonzz-96) [reply src/components/VideoCard/styled.js:29]: Well, I copied the design from the challenge demo. 😅

File: src/pages/Home/Home.page.jsx (hunk 3 above)
Comment (Gonzz-96) [reply src/pages/Home/Home.page.jsx]: Well, I made it this way just to have more test coverage. 😅  I didn't like how it looked either. I'll correct it. :)

**Comment by Gonzz-96** [comment]
Thanks @PacoMojica for your observations! 🎉

**Pull request merged**